server/game/cards/01_SOR/events/SmokeAndCinders.ts

```diff
@@ -0,0 +1,27 @@
+import AbilityHelper from '../../../AbilityHelper';
+import { EventCard } from '../../../core/card/EventCard';
+
+export default class SmokeAndCinders extends EventCard {
+    protected override getImplementationId() {
+        return {
+            id: '6702266551',
+            internalName: 'smoke-and-cinders',
+        };
+    }
+
+    public override setupCardAbilities() {
+        this.setEventAbility({
+            title: 'Each player discards all but 2 cards from their hand',
+            immediateEffect: AbilityHelper.immediateEffects.simultaneous(
+                (context) => context.game.getPlayers().map((player) =>
+                    AbilityHelper.immediateEffects.discardCardsFromOwnHand({
```

Comment:
This is a really clever way to pull this off, I think it might be better though if we could add something that directly supports this pattern as a feature for the sake of people who might have trouble parsing this level of complexity.

We had a false start on doing a Qira leader implementation a while back (long story), but we added there an option to the damage system for amount to be a function that accepts a card that could be computed dynamically. I think it would be good to do the same for the discard system as it would read much easier (though in this case it would accept a player obviously): https://github.com/SWU-Karabast/forceteki/blob/51a7527b4197176f3aaed36c31786979278c0ce6/server/game/gameSystems/DamageSystem.ts#L10